src/services/content.ts

```diff
@@ -0,0 +1,344 @@
+  blockIndex: number,
+  batchIndex: number
+) => {
+  if (isActivityContentNoteType(activityContent)) {
+    return dispatchFeedItem(
+      dispatch,
+      message,
+      activityContent as ActivityContentNote,
+      blockNumber
+    );
+  } else if (isActivityContentProfileType(activityContent)) {
+    return dispatchProfile(
+      dispatch,
+      message,
+      activityContent as ActivityContentProfile,
+      blockNumber,
+      blockIndex,
+      batchIndex
+    );
+  }
+};
+
+/**
+ * dispatchFeedItem dispatches a feed item to the redux store
+ * @param dispatch function used to dispatch to store
+ * @param message announcment information from batch
+ * @param content activity content retrieved from announcment
+ * @param blockNumber number of block containing the publication
+ */
+const dispatchFeedItem = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement | ReplyAnnouncement,
+  content: ActivityContentNote,
+  blockNumber: number
+) => {
+  const decoder = new TextDecoder();
+  dispatch(
+    addFeedItem({
+      fromId: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber: blockNumber,
+      hash: decoder.decode((message.contentHash as any) as Uint8Array),
+      published: content.published,
+      uri: decoder.decode((message.url as any) as Uint8Array),
+      content: content,
+      inReplyTo:
+        message.announcementType === core.announcements.AnnouncementType.Reply
+          ? decoder.decode((message.inReplyTo as any) as Uint8Array)
+          : undefined,
+    })
+  );
+};
+
+/**
+ * dispatchProfile dispatches a profile to the redux store
+ * @param dispatch function used to dispatch to store
+ * @param message announcment information from batch
+ * @param profile profile to dispatch
+ * @param blockNumber number of block containing the publication
+ * @param blockIndex index of log message (relative to other DSNP logs) within the block
+ * @param batchIndex index of announcment within the batch
+ */
+const dispatchProfile = (
+  dispatch: Dispatch,
+  message: BroadcastAnnouncement,
+  profile: ActivityContentProfile,
+  blockNumber: number,
+  blockIndex: number,
+  batchIndex: number
+) => {
+  const decoder = new TextDecoder();
+
+  dispatch(
+    upsertProfile({
+      ...profile,
+      fromId: decoder.decode((message.fromId as any) as Uint8Array),
+      blockNumber,
+      blockIndex,
+      batchIndex,
+    } as Profile)
+  );
+};
+
+/**
+ * handleRegistryUpdate dispatches a profile update for the handle when a registry update is made.
+ * @param dispatch function used to dispatch to store
+ * @param update registry log message containg DSNP uri and handle
+ */
+const handleRegistryUpdate = (dispatch: Dispatch) => (
+  update: RegistryUpdateLogData
+) => {
+  dispatch(
+    upsertProfile({
+      ...createProfile(),
+      fromId: core.identifiers.convertDSNPUserURIToDSNPUserId(
+        update.dsnpUserURI
+      ),
+      handle: update.handle,
+    } as Profile)
```

Comment:
Why do we need the `as`?

Reply:
This is resolved in the SDK 3.0 PR. It's due to a mismatch in types, which wasn't a good idea.